### ISSUE TITLE
Possible regression from email parseable in 0.35.3

### ISSUE THREAD

**Issue by shamusfield-expel** [TeamMsgExtractor/msg-extractor#423]
**Bug Metadata**
* Version of extract_msg: 0.35.3 -> 0.48.5/0.48.7
* Your python version: Python 3.9.2
* How did you launch extract_msg? extract-msg package

**Describe the bug**

Hello, I've got an interesting issue. I recently updated my package version from `0.35.3` to `0.48.5`, and an email that was previously able to be parsed is no longer able to be. The exception comes from the `olefile` library, however that version remains static at `0.47`. The exception is below. Under 0.35.3, no exception is produced and all the information I want (headers, body, attachments) appears to be intact and correct.

**What code did you use or can we use to reproduce this error?

`msg_parsed = extract_msg.Message(file_contents)`

**Is there a message.msg file you want to share to help us reproduce this?**

There is a msg file I have that causes this, but I'm unfortunately not allowed to provide it. That's probably as frustrating to you as it is to me, but thems the rules. 

**Traceback**

```
    msg_parsed = extract_msg.Message(file_contents)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/msg_classes/message_base.py", line 82, in __init__
    super().__init__(path, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/msg_classes/msg.py", line 221, in __init__
    self.attachments
  File "/usr/lib/python3.9/functools.py", line 969, in __get__
    val = self.func(instance)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/msg_classes/msg.py", line 862, in attachments
    attachments.append(self.initAttachmentFunc(self, attachmentDir))
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/attachments/__init__.py", line 103, in initStandardAttachment
    return Attachment(msg, dir_, propStore)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/attachments/attachment.py", line 39, in __init__
    super().__init__(msg, dir_, propStore)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/attachments/attachment_base.py", line 56, in __init__
    self.__namedProperties = NamedProperties(msg.named, self)
  File "/usr/lib/python3.9/functools.py", line 969, in __get__
    val = self.func(instance)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/msg_classes/msg.py", line 1034, in named
    return Named(self)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/properties/named.py", line 52, in __init__
    guidStream = self.getStream('__substg1.0_00020102')
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/properties/named.py", line 237, in getStream
    return msg.getStream([self.__dir, msgPathToString(filename)], False)
  File "/usr/local/lib/python3.9/dist-packages/extract_msg/msg_classes/msg.py", line 692, in getStream
    with self.__ole.openstream(filename) as stream:
  File "/usr/local/lib/python3.9/dist-packages/olefile/olefile.py", line 1970, in openstream
    return self._open(entry.isectStart, entry.size)
  File "/usr/local/lib/python3.9/dist-packages/olefile/olefile.py", line 1867, in _open
    return OleStream(fp=self.ministream, sect=start, size=size,
  File "/usr/local/lib/python3.9/dist-packages/olefile/olefile.py", line 649, in __init__
    self.ole._raise_defect(DEFECT_INCORRECT, 'incorrect OLE sector index for empty stream')
  File "/usr/local/lib/python3.9/dist-packages/olefile/olefile.py", line 1166, in _raise_defect
    raise exception_type(message)
olefile.olefile.OleFileError: incorrect OLE sector index for empty stream
```

**Additional context**

File contents in the code provided is a get_content() call in the base python email library. In this case we're parsing an eml file with that python lib, and then if we find an msg attachment in there, parsing that further with this library. I've tried both 0.48.5 and 0.48.7, both throw exceptions that the old version does not. 

I realize there's not a lot of information to go off of here, and without the email there's probably only so much you can do. I'm not even sure this is really a bug vs say, something being permitted in the old version that shouldn't have been. There's a lot of diff between those versions. Mostly what I'm hoping for is a possibility that either you know off hand something like "oh yeah we stopped allowing `<x>` in this particular version" or perhaps I'll get lucky and you happen to know what the issue is. I'm not particularly familiar with msg, so I don't think there's a feasible way for me to recreate an example.

Let me know what you think, thanks!

**Comment by TheElementalOfDestruction** [comment]
From a quick glance, my guess is that this has to do with a change to the level of error severity from the olefile library (see the section in extract_msg/msg_classes/msg.py line 151

The code was changed to make it so that more msg issues coming from the olefile level would, by default, throw errors to be identified, where before *no* options existed to enable this behavior, sometimes making confusion as to where an issue actually originated. This can be reverted to suppress these errors by adding the `ErrorBehavior.OLE_DEFECT_INCORRECT` to the flags specified in the `errorBehavior` argument given to openMsg. Passing that argument means that only fatal errors are reported.

There is a high likelihood that you seeing errors means there is something pretty badly wrong with the file you have, but not seeing it in the original version means it's not quite bad enough to break everything on the olefile level.

**Comment by TheElementalOfDestruction** [comment]
This change was made in version 0.42.0

**Comment by shamusfield-expel** [comment]
Hey, thanks for the quick response! Been tinkering a bit, and can confirm it was broken in 0.42.0, the prior version works. That flag allows it to continue, but doesn't produce similar results, which is curious. Prior to 0.42.0, I can get some mimetypes on attachments 
```
image/jpeg
application/pdf
```

but on 0.42.0 they're both `None`, so I'm guessing it's related to those attachments in some way. Also interesting is that it appears only the content types are not working, everything else (body, headers, attachment filenames & data) is exactly the same. I'm going to let my brain waffle on the best way to solve this, I may call it close enough and throw the data against a file type lib. In case that particular detail helps, awesome, if not, feel free to close and I can re-open if I find anything more detailed on my end or in the unlikely event I can share the msg.

Thanks a lot for the help/direction!

**Comment by TheElementalOfDestruction** [comment]
Does the 0.41.0 release work as you would expect? If so, then we can blame something between (0.41.0, 0.42.0] and try to figure out what changed to cause that issue. Not sure what changed that would have prevented parsing on some of those.

Anyways, the addition of one of our optional libraries will generate the content type from the actual data, so you can simply install that. The library is the python-magic library. Though keep in mind that its installation will suppress the ability to track down the specific version where things stop working.

Keep me updated

**Comment by shamusfield-expel** [comment]
> Does the 0.41.0 release work as you would expect? If so, then we can blame something between (0.41.0, 0.42.0] and try to figure out what changed to cause that issue. Not sure what changed that would have prevented parsing on some of those.

`0.41.5` works, `0.4.20` doesn't, so pretty confident it was the 0.4.20 release. 

> Anyways, the addition of one of our optional libraries will generate the content type from the actual data, so you can simply install that. The library is the python-magic library. Though keep in mind that its installation will suppress the ability to track down the specific version where things stop working.

I've actually got that in use elsewhere, so that will be a good fallback. having those types isn't the end of the world, for my use case, much better to continue with what we've got than stop on error.

**Comment by TheElementalOfDestruction** [comment]
Good to know, I'll try to take a look. Keep in mind that the 0.42.0 release was *massive* and I've got some other things I'm trying to track down so it may take me a bit of time.

**Comment by shamusfield-expel** [comment]
> Good to know, I'll try to take a look. Keep in mind that the 0.42.0 release was massive and I've got some other things I'm trying to track down so it may take me a bit of time.

No worries at all; thanks again for your help! I'll post back if I learn anything else or can come up with an example etc

**Comment by TheElementalOfDestruction** [comment]
So I went and did some digging, and I'm not surprised at how broken the mimetype is in 0.42.0 (it looks like I did the arguments for an internal method entirely wrong), but there shouldn't be any issues with acquiring it. I did find a bug that wouldn't affect the acquisition though, but logging should be able to tell me a lot more.

If you could run the following code with the failing files and let me know the results for 0.41.5 and 0.48.7, I'd really appreciate it since it should tell me the difference between the two. This will also not reveal *any* sensitive data from the msg files, and you can double check that if you want to be sure before sending it.

```py
import logging
import sys

import extract_msg

# Setup logging to the console.
logging.basicConfig(
    level=logging.INFO,
    format="[%(levelname)s] %(message)s",
    handlers=[logging.StreamHandler(sys.stdout)],
)

with extract_msg.openMsg('path/to/msg/file.msg') as msg:
    # This is here to let me know where to start my search for issues.
    logging.info("HERE WE ARE")
    # Attempt 
    for index, att in enumerate(msg.attachments):
        # Print the attachment number to better see if the problem is specific.
        logging.info(f'Attachment {index}')
        # Fetch it and store it so it's not logged to the console.
        temp = att.mimetype
```
The information I need will all be logged to the console. Thanks.

**Comment by shamusfield-expel** [comment]
`0.41.5`:

```
[INFO] Header found, but "cc" is not included. Will be generated from other streams.
[INFO] Stream "__substg1.0_1000001F" was requested but could not be found. Returning `None`.
[INFO] Stream "__substg1.0_10090102" was requested but could not be found. Returning `None`.
[INFO] HERE WE ARE
[INFO] Attachment 0
[INFO] Stream "__attach_version1.0_#00000000/__substg1.0_370E001F" was requested but could not be found. Returning `None`.
[INFO] image/jpeg
[INFO] Attachment 1
[INFO] Stream "__attach_version1.0_#00000001/__substg1.0_370E001F" was requested but could not be found. Returning `None`.
[INFO] application/pdf
```

`0.48.7`:
```
[INFO] Stream "__substg1.0_1000001F" was requested but could not be found. Returning `None`.
[INFO] Stream "__substg1.0_10090102" was requested but could not be found. Returning `None`.
[INFO] HERE WE ARE
[INFO] Attachment 0
[INFO] Stream "__attach_version1.0_#00000000/__substg1.0_370E001F" was requested but could not be found. Returning `None`.
[INFO] image/jpeg
[INFO] Attachment 1
[INFO] Stream "__attach_version1.0_#00000001/__substg1.0_370E001F" was requested but could not be found. Returning `None`.
[INFO] application/pdf
```

So it seems pretty much the same, as long as the OLE_DEFECT_INCORRECT flag is included in the 0.48.7. I assumed you wanted the temp var logged so added that as well.

**Comment by shamusfield-expel** [comment]
Which is curious, because with that code, it does appear able to find them. Let me go look at that....

**Comment by TheElementalOfDestruction** [comment]
I actually didn't want the temp car logged (my comments said so) but curiously despite the stream not being found in *both* versions, both versions show that mimetype information was returned. That's likely because the presence of the dependency alone will cause it to try to determine the mimetype based on the attachment data.

But both versions are showing the exact same result: couldn't find the mimetype information normally so it was generated. I'm not seeing the reported issue there

**Comment by shamusfield-expel** [comment]
> I actually didn't want the temp car logged (my comments said so)

Oops was not awake at that moment in time lol

> But both versions are showing the exact same result: couldn't find the mimetype information normally so it was generated. I'm not seeing the reported issue there. It's possible the issue is somehow presenting when the optional dependency is not present.

Yeah, for the life of me I can't seem to recreate those `None` results on mimetypes. I know they were there, as I saved a bunch of my console output the other day, but I can't figure out why. One thing I did as part of my bugfix release with this change was remove the `file-magic` lib from my setup.py since it was unused (`python-magic` still in place), and I'm wondering if maybe the old docker container had something different as a result of having that as well? I'm not sure, I've tried putting it back without difference. I've also tried putting the docker build lib back to `0.35.3` and manually pip installing the later versions, as well as a pile of other variations, to see if it was something like that but no dice. At this point I think I'm going to chalk that up to some sort of environment issue.

It's frustrating though, as I could flip back and forth and see it consistently over an over last week, so I'm not sure exactly what changed. Apologies for the wild goose chase, and if I can manage to recreate I'll let you know. For now seems fine though 🤷
